internal/server/storage/drivers/utils_ceph.go

```diff
@@ -208,20 +211,119 @@ func CephKeyring(cluster string, client string) (string, error) {
 	key := struct {
 		Key string `json:"key"`
 	}{}
-	err = callCephJSON(&key,
-		"--cluster", cluster,
-		"auth", "get-key", client,
-	)
+	err = callCephJSON(&key, "--cluster", cluster, "auth", "get-key", client)
 	if err != nil {
-		return "", fmt.Errorf(
-			"Failed to get keyring for %q on %q: %w",
-			client, cluster, err,
-		)
+		return "", fmt.Errorf("Failed to get keyring for %q on %q: %w", client, cluster, err)
 	}
 
 	return key.Key, nil
 }
 
+func cephGetKeyFromFile(path string) (string, error) {
+	cephKeyring, err := os.Open(path)
+	if err != nil {
+		return "", fmt.Errorf("Failed to open %q: %w", path, err)
+	}
+
+	// Locate the keyring entry and its value.
+	var cephSecret string
+	scan := bufio.NewScanner(cephKeyring)
+	for scan.Scan() {
+		line := scan.Text()
+		line = strings.TrimSpace(line)
+
+		if line == "" {
+			continue
+		}
+
+		if strings.HasPrefix(line, "key") {
+			fields := strings.SplitN(line, "=", 2)
```

Comment:
here and below, it seems like you should first drop anything after a # or ;. Else a line like

```
key = value # used to be key = other
```

will wrongly be skipped.

Reply:
I'm need to review the logic again, for now what's in there is a straight up revert to the logic we've had for the past 5+ years so probably best to go with that for the regression fix and then look at it again later

Reply:
Sounds good.